base/pkg/libgit2/types.jl

```diff
@@ -0,0 +1,380 @@
+                        notify_flags,
+                        notify_cb,
+                        notify_payload,
+                        progress_cb,
+                        progress_payload,
+                        paths,
+                        baseline,
+                        target_directory,
+                        ancestor_label,
+                        our_label,
+                        their_label)
+
+immutable RemoteCallbacksStruct
+    version::Cuint
+    sideband_progress::Ptr{Void}
+    completion::Ptr{Void}
+    credentials::Ptr{Void}
+    certificate_check::Ptr{Void}
+    transfer_progress::Ptr{Void}
+    update_tips::Ptr{Void}
+    pack_progress::Ptr{Void}
+    push_transfer_progress::Ptr{Void}
+    push_update_reference::Ptr{Void}
+    payload::Ptr{Void}
+end
+RemoteCallbacksStruct() = RemoteCallbacksStruct(one(Cuint),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0),
+                                                Ptr{Void}(0))
+
+immutable CloneOptionsStruct
+    version::Cuint
+    checkout_opts::CheckoutOptionsStruct
+    remote_callbacks::RemoteCallbacksStruct
+    bare::Cint
+    localclone::Cint
+    checkout_branch::Ptr{UInt8}
+    signature::Ptr{Void}
+    repository_cb::Ptr{Void}
+    repository_cb_payload::Ptr{Void}
+    remote_cb::Ptr{Void}
+    remote_cb_payload::Ptr{Void}
+end
+CloneOptionsStruct(; checkout_opts::CheckoutOptionsStruct = CheckoutOptionsStruct(),
+                     remote_callbacks::RemoteCallbacksStruct = RemoteCallbacksStruct(),
+                     bare::Cint = zero(Cint),
+                     localclone::Cint = zero(Cint),
+                     checkout_branch::Ptr{UInt8} = Ptr{UInt8}(0),
+                     signature::Ptr{Void} = Ptr{Void}(0),
+                     repository_cb::Ptr{Void} = Ptr{Void}(0),
+                     repository_cb_payload::Ptr{Void} = Ptr{Void}(0),
+                     remote_cb::Ptr{Void} = Ptr{Void}(0),
+                     remote_cb_payload::Ptr{Void} = Ptr{Void}(0)
+)=CloneOptionsStruct(one(Cuint),
+                     checkout_opts,
+                     remote_callbacks,
+                     bare,
+                     localclone,
+                     checkout_branch,
+                     signature,
+                     repository_cb,
+                     repository_cb_payload,
+                     remote_cb,
+                     remote_cb_payload)
+
+# git diff option struct
+immutable DiffOptionsStruct
```

Comment:
Did you forget to define `Base.finalize` for this?  (The finalizer needs to finalize `pathspec`.)